crates/re_edit_ui/src/visual_bounds2d.rs

```diff
@@ -0,0 +1,153 @@
+    let speed_func = |start: f64, end: f64| ((end - start).abs() * 0.01).at_least(0.001);
+
+    let mut any_edit = false;
+
+    let response_x = re_ui::list_item::ListItem::new(ctx.re_ui)
+        .interactive(false)
+        .show_flat(
+            ui,
+            re_ui::list_item::PropertyContent::new("x").value_fn(|_, ui, _| {
+                let [x_range_start, x_range_end] = &mut value.x_range.0;
+                let speed = speed_func(*x_range_start, *x_range_end);
+
+                let response = ui
+                    .horizontal_centered(|ui| {
+                        let response_min = ui.add(
+                            egui::DragValue::new(x_range_start)
+                                .clamp_range(f64::MIN..=*x_range_end)
+                                .max_decimals(2)
+                                .speed(speed),
+                        );
+
+                        ui.label("-");
+
+                        let response_max = ui.add(
+                            egui::DragValue::new(x_range_end)
+                                .clamp_range(*x_range_start..=f64::MAX)
+                                .max_decimals(2)
+                                .speed(speed),
+                        );
+
+                        response_min | response_max
+                    })
+                    .inner;
+
+                if response.changed() {
+                    any_edit = true;
+                }
+            }),
+        );
+
+    let response_y = re_ui::list_item::ListItem::new(ctx.re_ui)
+        .interactive(false)
+        .show_flat(
+            ui,
+            re_ui::list_item::PropertyContent::new("y").value_fn(|_, ui, _| {
+                let [y_range_start, y_range_end] = &mut value.y_range.0;
+                let speed = speed_func(*y_range_start, *y_range_end);
+
+                let response = ui
+                    .horizontal_centered(|ui| {
+                        let response_min = ui.add(
+                            egui::DragValue::new(y_range_start)
+                                .clamp_range(f64::MIN..=*y_range_end)
+                                .max_decimals(2)
+                                .speed(speed),
+                        );
+
+                        ui.label("-");
+
+                        let response_max = ui.add(
+                            egui::DragValue::new(y_range_end)
+                                .clamp_range(*y_range_start..=f64::MAX)
+                                .max_decimals(2)
+                                .speed(speed),
+                        );
+
+                        response_min | response_max
+                    })
+                    .inner;
+
+                if response.changed() {
+                    any_edit = true;
+                }
+            }),
+        );
+
+    let mut response = response_x | response_y;
+    if any_edit {
+        response.mark_changed();
+    }
+    response
+}
+
+pub fn singleline_edit_visual_bounds2d(
```

Comment:
Here I'm wondering if whether we can is obscuring whether we should :) The UX is not bad, but a bit surprising. Let's see what people think about it—I'd remove it if it induces too much confusion.